### ISSUE TITLE
Memory leak with keras metrics

### ISSUE THREAD

**Issue by jnd77** [tensorflow/tensorflow#44756]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): tf-nightly
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**

The following test is failing, as the metric is not properly garbage collected.

```
import tensorflow as tf
from tensorflow.python.framework.test_util import assert_no_garbage_created

class TestMetric(tf.test.TestCase):
  @assert_no_garbage_created
  def test_metric(self):
    metric = tf.keras.metrics.Mean('test', dtype=tf.float32)

import unittest
unittest.main(argv=['first-arg-is-ignored'], exit=False)
```

**Describe the expected behavior**

Test should pass

**Standalone code to reproduce the issue**
See above.

**Other info / logs**

**Comment by Saduf2019** [comment]
@jnd77 
I ran the code shared on tf 2.3 and nightly, please let us know if this confirm your [issue](https://colab.research.google.com/gist/Saduf2019/0d4cc5e30f0dddd8efefdec172a49e96/untitled457.ipynb).

**Comment by jnd77** [comment]
@Saduf2019 
Thanks for reproducing it. It confirms the issue.

**Comment by ymodak** [comment]
This is fixed with latest tf-nightly. See colab [gist](https://colab.research.google.com/gist/ymodak/13256d6196cff01519be16827b321333/untitled457.ipynb) to know more. Thanks!
